distsql/stream.go

```diff
@@ -43,10 +43,15 @@ type streamResult struct {
 
 func (r *streamResult) Fetch(context.Context) {}
 
+// Next reads data to the chunk.
 func (r *streamResult) Next(ctx context.Context, chk *chunk.Chunk) error {
-	chk.Reset()
-	maxChunkSize := r.ctx.GetSessionVars().MaxChunkSize
-	for chk.NumRows() < maxChunkSize {
+	return r.NextBatch(ctx, chunk.NewRecordBatch(chk))
```

Comment:
how about setting the required rows of that newly created batch to max chunk size?